pkg/server/container_create_windows.go

```diff
@@ -166,8 +171,28 @@ func (c *criService) CreateContainer(ctx context.Context, r *runtime.CreateConta
 
 	log.G(ctx).Debugf("Container %q spec: %#+v", id, spew.NewFormatter(spec))
 
-	snapshotterOpt := snapshots.WithLabels(config.Annotations)
+	// If the config field is specified, set the snapshotter label to reuse the pods
+	// scratch space. This only affects LCOW at the moment.
+	val, ok := config.Annotations["containerd.io/snapshot/io.microsoft.container.storage.reuse-scratch"]
+	if rhcso.ShareScratch || (ok && val == "true") {
+		config.Annotations["containerd.io/snapshot/io.microsoft.container.storage.reuse-scratch"] = "true"
+		_, ok := config.Annotations["containerd.io/snapshot/io.microsoft.owner.key"]
+		if !ok {
+			config.Annotations["containerd.io/snapshot/io.microsoft.owner.key"] = sandboxID
+		}
+	}
 
+	if rhcso.DefaultContainerScratchSizeInGb != 0 {
```

Comment:
I dont remember how this works. Can you override the default opts in the config.toml is why we check here? Basically why are we forwarding a default value versus just "not passing" if unset?

Reply:
I think if you don't supply the config option at all it would just unmarshal as the default for whatever type it is, so if this option was omitted for the LCOW runtime in containerd.toml this block would be skipped. Of course you could always define it and set it to 0 anyways in which case we'd also hit this but I think that's fine, not sure what a 0GB scratch space would look like 😆

Reply:
Ah ok np